SiStrip (un)packer: fixes and support for nonstandard ZS(lite) modes

pieterdavid opened the pull request:
This PR contains fixes for some bugs in the SiStrip unpacker for 10bit ZS(lite) readout modes (found by @erikbutz when testing the new hybrid zero suppression for heavy ion data taking later this year), and adds support for all different packing schemes (readout modes and packet codes, many were missing) to the packer (used in simulation and to repack HI data in the HLT).
The unpacker changes are limited to the code used for 10bit packed data (plus the code in the EDProducer to add cases for the modes that were not taken into account there), so there should be no effect on what is used for normal pp data. The packer changes were tested by unpacking its output and comparing the digis to the original collections, taking into account the expected differences (see [this config](https://github.com/pieterdavid/SiStripLocalRecoDebugging/blob/master/test/testPacker_fromVR.py) and related code).

with @alesaggio 
CC: @OlivierBondu @vidalm  @mmusich @echabert

pieterdavid committed:
Fix packet codes for ZS and ZS lite

- rename the PACKET_CODE definitions for ZS lite to ZS
- change the packetCode implementation: default (0) for ZS lite modes,
  taken from the third byte for ZS
Reason: for ZS lite there are no packet codes, the packing scheme
is given by the readout mode; for non-lite ZS it is in the channel data

pieterdavid committed:
Add other ZS modes to SiStripRawToDigiUnpacker

pieterdavid committed:
Fix uninitialized member in FEDBSChannelUnpacker

pieterdavid committed:
A few more FEDBSChannelUnpacker fixes

pieterdavid committed:
Add packet code option for SiStripDigiToRawModule

- add "PacketCode" option to SiStripDigiToRawModule
- add packetCodeFromString conversion in SiStripFEDBufferComponents
- pass the packet code on to FEDBufferPayloadCreator::fillChannelBuffer

alesaggio committed:
adding VR packet code cases

pieterdavid committed:
Fix a regression for VR10 unpacking

pieterdavid committed:
use an InputTag in SiStripDigiToRawModule

pieterdavid committed:
Fix FEDBSChannelUnpacker::hasData for other sizes than 10 and 16

alesaggio committed:
VR and ZR fixed

pieterdavid committed:
Merge branch 'SiStripPacker_fixZSmodes_102' of https://github.com/alesaggio/cmssw into SiStripUnpacker_fixZSmodes_102

pieterdavid committed:
ZS packer patch: cleanup

pieterdavid committed:
Packer: use ZS code also for ZS lite

pieterdavid committed:
packer: readout mode parsing and empty default packet code

pieterdavid committed:
also for ZS packer: push directly packet code

pieterdavid committed:
code checks fixes (empty string tests)

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-23417/4957

cmsbuild commented:
A new Pull Request was created by  @pieterdavid (Pieter David) for master.

It involves the following packages:

EventFilter/SiStripRawToDigi


@perrotta, @cmsbuild, @slava77 can you please review it and eventually sign? Thanks.
@makortel, @Martin-Grunewald, @VinInn this is something you requested to watch as well.
@davidlange6, @slava77, @fabiocos you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

mmusich commented:
@icali FYI

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: 983db858708a77e601d40509f75169e6ed771b62


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/86a9c4c3025c89150d0623faf6034a298900750d
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23417/28368/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23417/28368/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23417/28368/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals  AddOn**

* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
8.0 step2
<pre>runTheMatrix-results/8.0_BeamHalo+BeamHalo+DIGICOS+RECOCOS+ALCABH+HARVESTCOS/step2_BeamHalo+BeamHalo+DIGICOS+RECOCOS+ALCABH+HARVESTCOS.log

</pre>7.3 step2
<pre>runTheMatrix-results/7.3_CosmicsSPLoose_UP17+CosmicsSPLoose_UP17+DIGICOS_UP17+RECOCOS_UP17+ALCACOS_UP17+HARVESTCOS_UP17/step2_CosmicsSPLoose_UP17+CosmicsSPLoose_UP17+DIGICOS_UP17+RECOCOS_UP17+ALCACOS_UP17+HARVESTCOS_UP17.log

</pre>9.0 step2
<pre>runTheMatrix-results/9.0_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST/step2_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST.log

</pre>25.0 step2
<pre>runTheMatrix-results/25.0_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT/step2_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT.log

</pre>1306.0 step2
<pre>runTheMatrix-results/1306.0_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15/step2_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>150.0 step2
<pre>runTheMatrix-results/150.0_HydjetQ_B12_5020GeV_2018+HydjetQ_B12_5020GeV_2018+DIGIHI2018+RECOHI2018+HARVESTHI2018/step2_HydjetQ_B12_5020GeV_2018+HydjetQ_B12_5020GeV_2018+DIGIHI2018+RECOHI2018+HARVESTHI2018.log

</pre>1330.0 step2
<pre>runTheMatrix-results/1330.0_ZMM_13+ZMM_13+DIGIUP15+RECOUP15_L1TMuDQM+HARVESTUP15_L1TMuDQM/step2_ZMM_13+ZMM_13+DIGIUP15+RECOUP15_L1TMuDQM+HARVESTUP15_L1TMuDQM.log

</pre>25202.0 step2
<pre>runTheMatrix-results/25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25/step2_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25.log

</pre>10042.0 step2
<pre>runTheMatrix-results/10042.0_ZMM_13+ZMM_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017+ALCAFull_2017/step2_ZMM_13+ZMM_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017+ALCAFull_2017.log

</pre>10024.0 step2
<pre>runTheMatrix-results/10024.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017+ALCAFull_2017/step2_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017+ALCAFull_2017.log

</pre>10224.0 step2
<pre>runTheMatrix-results/10224.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017PU_GenSimFull+DigiFullPU_2017PU+RecoFullPU_2017PU+HARVESTFullPU_2017PU/step2_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017PU_GenSimFull+DigiFullPU_2017PU+RecoFullPU_2017PU+HARVESTFullPU_2017PU.log

</pre>10824.0 step2
<pre>runTheMatrix-results/10824.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_2018+HARVESTFull_2018+ALCAFull_2018/step2_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_2018+HARVESTFull_2018+ALCAFull_2018.log

</pre>11624.0 step2
<pre>runTheMatrix-results/11624.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2019_GenSimFull+DigiFull_2019+RecoFull_2019+HARVESTFull_2019+ALCAFull_2019/step2_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2019_GenSimFull+DigiFull_2019+RecoFull_2019+HARVESTFull_2019+ALCAFull_2019.log

</pre>20034.0 step2
<pre>runTheMatrix-results/20034.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D17_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D17+RecoFullGlobal_2023D17+HARVESTFullGlobal_2023D17/step2_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D17_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D17+RecoFullGlobal_2023D17+HARVESTFullGlobal_2023D17.log

</pre>250202.181 step2
<pre>runTheMatrix-results/250202.181_TTbar_13UP18+TTbar_13UP18+PREMIXUP18_PU25+DIGIPRMXLOCALUP18_PU25+RECOPRMXUP18_PU25+HARVESTUP18_PU25/step2_TTbar_13UP18+TTbar_13UP18+PREMIXUP18_PU25+DIGIPRMXLOCALUP18_PU25+RECOPRMXUP18_PU25+HARVESTUP18_PU25.log

</pre>21234.0 step2
<pre>runTheMatrix-results/21234.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D21_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D21+RecoFullGlobal_2023D21+HARVESTFullGlobal_2023D21/step2_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D21_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D21+RecoFullGlobal_2023D21+HARVESTFullGlobal_2023D21.log

</pre>20434.0 step2
<pre>runTheMatrix-results/20434.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D19_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D19+RecoFullGlobal_2023D19+HARVESTFullGlobal_2023D19/step2_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D19_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D19+RecoFullGlobal_2023D19+HARVESTFullGlobal_2023D19.log

</pre>

* **AddOn**:

I found errors in the following addon tests:

cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=pp -n 10 --conditions auto:run2_mc_GRun --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2018 --fileout file:RelVal_Raw_GRun_MC.root : FAILED - time: date Fri Jun  1 18:00:35 2018-date Fri Jun  1 17:58:41 2018 s - exit: 18688
cmsRun /cvmfs/cms-ib.cern.ch/nweek-02526/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_2_X_2018-05-31-2300/src/HLTrigger/Configuration/test/OnLine_HLT_GRun.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Fri Jun  1 18:00:35 2018-date Fri Jun  1 17:58:41 2018 s - exit: 21504
cmsDriver.py RelVal -s HLT:GRun,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_GRun --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2018 --processName=HLTRECO --filein file:RelVal_Raw_GRun_MC.root --fileout file:RelVal_Raw_GRun_MC_HLT_RECO.root : FAILED - time: date Fri Jun  1 18:00:35 2018-date Fri Jun  1 17:58:41 2018 s - exit: 21504
cmsDriver.py TTbar_Tauola_8TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW  --mc --scenario=pp -n 10 --conditions auto:run1_mc_Fake --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --fileout file:RelVal_Raw_Fake_MC.root : FAILED - time: date Fri Jun  1 17:59:49 2018-date Fri Jun  1 17:58:46 2018 s - exit: 18688
cmsRun /cvmfs/cms-ib.cern.ch/nweek-02526/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_2_X_2018-05-31-2300/src/HLTrigger/Configuration/test/OnLine_HLT_Fake.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Fri Jun  1 17:59:49 2018-date Fri Jun  1 17:58:46 2018 s - exit: 21504
cmsDriver.py RelVal -s HLT:Fake,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run1_mc_Fake --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --processName=HLTRECO --filein file:RelVal_Raw_Fake_MC.root --fileout file:RelVal_Raw_Fake_MC_HLT_RECO.root : FAILED - time: date Fri Jun  1 17:59:49 2018-date Fri Jun  1 17:58:46 2018 s - exit: 21504
cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=pp -n 10 --conditions auto:run2_mc_PRef --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2018 --fileout file:RelVal_Raw_PRef_MC.root : FAILED - time: date Fri Jun  1 18:01:22 2018-date Fri Jun  1 17:59:50 2018 s - exit: 18688
cmsRun /cvmfs/cms-ib.cern.ch/nweek-02526/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_2_X_2018-05-31-2300/src/HLTrigger/Configuration/test/OnLine_HLT_PRef.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Fri Jun  1 18:01:22 2018-date Fri Jun  1 17:59:50 2018 s - exit: 21504
cmsDriver.py RelVal -s HLT:PRef,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_PRef --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2018 --processName=HLTRECO --filein file:RelVal_Raw_PRef_MC.root --fileout file:RelVal_Raw_PRef_MC_HLT_RECO.root : FAILED - time: date Fri Jun  1 18:01:22 2018-date Fri Jun  1 17:59:50 2018 s - exit: 21504
cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=pp -n 10 --conditions auto:run2_mc_2018v21 --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2018 --fileout file:RelVal_Raw_2018v21_MC.root : FAILED - time: date Fri Jun  1 18:04:24 2018-date Fri Jun  1 18:01:31 2018 s - exit: 18688
cmsRun /cvmfs/cms-ib.cern.ch/nweek-02526/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_2_X_2018-05-31-2300/src/HLTrigger/Configuration/test/OnLine_HLT_2018v21.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Fri Jun  1 18:04:24 2018-date Fri Jun  1 18:01:31 2018 s - exit: 21504
cmsDriver.py RelVal -s HLT:2018v21,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_2018v21 --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2018 --processName=HLTRECO --filein file:RelVal_Raw_2018v21_MC.root --fileout file:RelVal_Raw_2018v21_MC_HLT_RECO.root : FAILED - time: date Fri Jun  1 18:04:24 2018-date Fri Jun  1 18:01:31 2018 s - exit: 21504
cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=HeavyIons -n 10 --conditions auto:run2_mc_HIon --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_HI --fileout file:RelVal_Raw_HIon_MC.root : FAILED - time: date Fri Jun  1 18:04:04 2018-date Fri Jun  1 18:02:26 2018 s - exit: 18688
cmsRun /cvmfs/cms-ib.cern.ch/nweek-02526/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_2_X_2018-05-31-2300/src/HLTrigger/Configuration/test/OnLine_HLT_HIon.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Fri Jun  1 18:04:04 2018-date Fri Jun  1 18:02:26 2018 s - exit: 21504
cmsDriver.py RelVal -s HLT:HIon,RAW2DIGI,L1Reco,RECO --mc --scenario=HeavyIons -n 10 --conditions auto:run2_mc_HIon --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_HI --processName=HLTRECO --filein file:RelVal_Raw_HIon_MC.root --fileout file:RelVal_Raw_HIon_MC_HLT_RECO.root : FAILED - time: date Fri Jun  1 18:04:04 2018-date Fri Jun  1 18:02:26 2018 s - exit: 21504
cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=pp -n 10 --conditions auto:run2_mc_Fake2 --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2016 --fileout file:RelVal_Raw_Fake2_MC.root : FAILED - time: date Fri Jun  1 18:04:24 2018-date Fri Jun  1 18:02:39 2018 s - exit: 18688
cmsRun /cvmfs/cms-ib.cern.ch/nweek-02526/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_2_X_2018-05-31-2300/src/HLTrigger/Configuration/test/OnLine_HLT_Fake2.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Fri Jun  1 18:04:24 2018-date Fri Jun  1 18:02:39 2018 s - exit: 21504
cmsDriver.py RelVal -s HLT:Fake2,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_Fake2 --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2016 --processName=HLTRECO --filein file:RelVal_Raw_Fake2_MC.root --fileout file:RelVal_Raw_Fake2_MC_HLT_RECO.root : FAILED - time: date Fri Jun  1 18:04:24 2018-date Fri Jun  1 18:02:39 2018 s - exit: 21504
cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=pp -n 10 --conditions auto:run2_mc_Fake1 --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_25ns --fileout file:RelVal_Raw_Fake1_MC.root : FAILED - time: date Fri Jun  1 18:05:04 2018-date Fri Jun  1 18:03:23 2018 s - exit: 18688
cmsRun /cvmfs/cms-ib.cern.ch/nweek-02526/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_2_X_2018-05-31-2300/src/HLTrigger/Configuration/test/OnLine_HLT_Fake1.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Fri Jun  1 18:05:04 2018-date Fri Jun  1 18:03:23 2018 s - exit: 21504
cmsDriver.py RelVal -s HLT:Fake1,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_Fake1 --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_25ns --processName=HLTRECO --filein file:RelVal_Raw_Fake1_MC.root --fileout file:RelVal_Raw_Fake1_MC_HLT_RECO.root : FAILED - time: date Fri Jun  1 18:05:04 2018-date Fri Jun  1 18:03:23 2018 s - exit: 21504
cmsDriver.py TTbar_Tauola_13TeV_TuneCUETP8M1_cfi -s GEN,SIM,DIGI,L1,DIGI2RAW --mc --scenario=pp -n 10 --conditions auto:run2_mc_PIon --relval 9000,50 --datatier "GEN-SIM-RAW" --eventcontent RAWSIM --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2018 --fileout file:RelVal_Raw_PIon_MC.root : FAILED - time: date Fri Jun  1 18:05:53 2018-date Fri Jun  1 18:04:39 2018 s - exit: 18688
cmsRun /cvmfs/cms-ib.cern.ch/nweek-02526/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_2_X_2018-05-31-2300/src/HLTrigger/Configuration/test/OnLine_HLT_PIon.py realData=False globalTag=@ inputFiles=@  : FAILED - time: date Fri Jun  1 18:05:53 2018-date Fri Jun  1 18:04:39 2018 s - exit: 21504
cmsDriver.py RelVal -s HLT:PIon,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_PIon --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2018 --processName=HLTRECO --filein file:RelVal_Raw_PIon_MC.root --fileout file:RelVal_Raw_PIon_MC_HLT_RECO.root : FAILED - time: date Fri Jun  1 18:05:53 2018-date Fri Jun  1 18:04:39 2018 s - exit: 21504



The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/86a9c4c3025c89150d0623faf6034a298900750d
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23417/28368/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23417/28368/git-merge-result

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

mmusich commented on EventFilter/SiStripRawToDigi/python/SiStripDigiToRaw_cfi.py:
```diff
@@ -5,6 +5,7 @@
     InputModuleLabel = cms.string('simSiStripDigis'),
     InputDigiLabel = cms.string('ZeroSuppressed'),
```
Hi @pieterdavid based on the relval errors across the board:
```
----- Begin Fatal Exception 01-Jun-2018 17:19:10 CEST-----------------------
An exception of category 'Configuration' occurred while
   [0] Constructing the EventProcessor
   [1] Validating configuration of module: class=SiStripDigiToRawModule label='SiStripDigiToRaw'
Exception Message:
Illegal parameters found in configuration.  The parameters are named:
 'InputDigiLabel'
 'InputModuleLabel'
You could be trying to use parameter names that are not
allowed for this plugin or they could be misspelled.
----- End Fatal Exception -------------------------------------------------
```
I think these two:
```
InputModuleLabel = cms.string('simSiStripDigis'),
InputDigiLabel = cms.string('ZeroSuppressed'),
```
should be gone as well to please the descriptions.

pieterdavid replied:
thanks @mmusich , indeed I forgot to fully update the configs for the packer... I am having a look at that (and running the matrix tests locally before pushing this time ;-) )

pieterdavid committed:
Update configs to changes in SiStripDigiToRaw parameters

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-23417/4971

cmsbuild commented:
Pull request #23417 was updated. @perrotta, @civanch, @mdhildreth, @cmsbuild, @franzoni, @slava77, @fabiocos, @davidlange6 can you please check and sign again.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: a980e349755027994f0086a4938c1bad65aec02e
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23417/28399/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/86a9c4c3025c89150d0623faf6034a298900750d
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23417/28399/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23417/28399/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23417/28399/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 308 differences found in the comparisons
 - DQMHistoTests: Total files compared: 31
 - DQMHistoTests: Total histograms compared: 2902471
 - DQMHistoTests: Total failures: 18681
 - DQMHistoTests: Total nulls: 6
 - DQMHistoTests: Total successes: 2883594
 - DQMHistoTests: Total skipped: 190
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: -3.198 KiB( 30 files compared)
 - DQMHistoSizes: changed ( 140.53 ): -3.198 KiB       SiStrip/MechanicalView
 - Checked 128 log files, 14 edm output root files, 31 DQM output files

pieterdavid commented:
If I read the outputs correctly, the failures in the comparisons seem to be mostly in the [140.53_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_10_2_X_2018-06-03-0000+23417/26863/140.53_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI/) workflow. Looking at the step2 logfile I do see quite some error messages, but not immediately something obviously strip-unpacker related, while there is a `grep: write error: Broken pipe` in the [step1_dasquery.log](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23417/28399/runTheMatrix-results/140.53_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI/step1_dasquery.log). Could it be a glitch in the tests? Or is there something else we missed?

mmusich commented:
Hi @pieterdavid, naively looking at the subfolders of the relmon summary for wf 140.53 there are no clusters at all with the changes proposed here:

https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_10_2_X_2018-06-03-0000+23417/26863/140.53_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI/SiStrip_MechanicalView_TOB.html

![image](https://user-images.githubusercontent.com/5082376/40978462-f9eff0cc-68d3-11e8-8e60-f4001d3a7934.png)



and consequently no tracks:

https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_10_2_X_2018-06-03-0000+23417/26863/140.53_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI/Tracking_TrackParameters_GeneralProperties.html

pieterdavid commented:
Thanks @mmusich, I'm running the test locally to try and debug

pieterdavid commented:
There is indeed something going wrong for 2011 HI data with these changes... what I implemented requires (non-lite) ZS data to have a valid packet code (and the same for ZS fake, because that's the only thing I found about it in the spec), otherwise the channel is ignored - but this data has readout mode `READOUT_MODE_ZERO_SUPPRESSED_FAKE` and packet code (third byte) zero.
Should I infer from this that ZS fake is always 8bit (topmost two bits truncated) and that the packet code should be ignored for it? Or should we take this packet code as default instead of ignoring the data? (there is a warning already now, but it's only printed in debug mode, in analogy with those from the surrounding code)

pieterdavid commented:
@icali pointed out that the 2011 HI files used in the test are most likely repacked, and that the above behaviour is due to a bug in the packer that was only fixed in 2015, with [this change](https://github.com/cms-sw/cmssw/commit/d0849639b5430352b16c3fb45dc7973e2e4a65ac), I think.
The question then is: should we implement a workaround, and what can it look like? I guess we want to keep the unpacker as strict as possible... we could add a boolean option (to fall back to the default packet code in case it is zero), and enable it only for the `run1_data` era (or only for this test)?

pieterdavid committed:
Use default ZS packing if the packet code is zero

workaround for a bug in the pre-2015 packer (eg. 140.53_RunHI2011)

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-23417/5062

cmsbuild commented:
Pull request #23417 was updated. @perrotta, @civanch, @mdhildreth, @cmsbuild, @franzoni, @slava77, @fabiocos, @davidlange6 can you please check and sign again.

pieterdavid commented:
With the commit I added the packer will use the default ZS packet code if it finds zero in the data - it is not configurable because I found a Run1 era but apparently it is not used (and also not set for this test)... please let me know if there is a better solution.

perrotta commented on EventFilter/SiStripRawToDigi/plugins/SiStripRawToDigiUnpacker.cc:
```diff
@@ -305,7 +305,13 @@ namespace sistrip {
                     << " for " << *ifed << " channel " << iconn->fedCh();
-                } }
+                }
+                if ( packet_code == 0 ) {
```
Is this the only possibility for packet_code being 0?
If so, the previous LogWarning shouldn't get fired by it, i.e.. that warning should only be put after an ``else if`` statement
If not, this workaround could lead to misbehaviours in case of ``packet_code == 0`` due to some different origin

pieterdavid replied:
From what I understood, packet code zero for readout mode zero suppressed is simply invalid (so there should be no other cases), which is why I would keep the warning (which is silenced unless debugging is on because it may print a message for every channel) even if we try to unpack such a channel anyway now, but that's only for backwards-compatibility with pre-2015 repacked data. It's easy to change to only issuing it `else if...`, though, if you think that's better.

pieterdavid replied:
Good idea, with the last commit I added a summary of warnings (this and the other errors that were silenced in non-debug mode, even though many are quite serious) that is printed once per event, in the case of the repacked 2011 HI data that becomes
```
%MSG-w SiStripRawToDigi:  SiStripRawToDigiModule:siStripDigis  07-Jun-2018 10:30:23 CEST Run: 182124 Event: 1893722
[sistrip::RawToDigiUnpacker::createDigis] warnings:
Invalid packet code 0 for zero-suppressed data (35592)
```
Would that work? (I'm sorry, I forgot about the "once per job" when implementing this... since these mean in principle (in all except this very specific case) that some data is ignored, once per event looks like a reasonable granularity, but it's straightforward to change)

pieterdavid committed:
Print SiStrip unpacker warnings and exceptions once per event

In debug mode all messages are still printed, otherwise a summary
(warning type and count) is printed after unpacking the event.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-23417/5086

cmsbuild commented:
Pull request #23417 was updated. @perrotta, @civanch, @mdhildreth, @cmsbuild, @franzoni, @slava77, @fabiocos, @davidlange6 can you please check and sign again.

perrotta commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

perrotta commented:
@pieterdavid : let launch the tests and see what happens...
But if really the warnings are issued "once per event", the flow of messages can become hige in case of issues. My suggestion was to issue them once per job (or thread) instead: for example, the first time they are encountered.

pieterdavid commented:
Indeed... I prepared the changes to print every warning the first time it is encountered, and the summary at the end of the job (from `endStream()`). @perrotta shall I push this commit now, or better wait for the tests to finish?

cmsbuild commented:
+1
Tested at: 57234aa9282cb8a5f7e4d8adf8115f63148e96b2
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23417/28533/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/86a9c4c3025c89150d0623faf6034a298900750d
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23417/28533/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23417/28533/git-merge-result

cmsbuild commented:
Comparison job queued.

perrotta commented:
Please Pieter, push it whenever you have it ready
We can re-launch the tests

Pieter David <notifications@github.com> ha scritto:

> Indeed... I prepared the changes to print every warning the first  
> time it is encountered, and the summary at the end of the job (from  
> `endStream()`). @perrotta shall I push this commit now, or better  
> wait for the tests to finish?
>
> --
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/cms-sw/cmssw/pull/23417#issuecomment-395378613

pieterdavid committed:
strip unpacker: print warnings on first occurrence, summary in endStream

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-23417/5092

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23417/28533/summary.html

Comparison Summary:
 - You potentially added 129 lines to the logs
 - Reco comparison results: 12 differences found in the comparisons
 - DQMHistoTests: Total files compared: 31
 - DQMHistoTests: Total histograms compared: 2902639
 - DQMHistoTests: Total failures: 2
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2902447
 - DQMHistoTests: Total skipped: 190
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 30 files compared)
 - Checked 128 log files, 14 edm output root files, 31 DQM output files

perrotta commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 06b4b3b5bbfeb465483334a8c52d66643fa93171
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23417/28542/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/86a9c4c3025c89150d0623faf6034a298900750d
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23417/28542/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23417/28542/git-merge-result

cmsbuild commented:
Comparison job queued.

perrotta commented on EventFilter/SiStripRawToDigi/interface/SiStripFEDBuffer.h:
```diff
@@ -270,13 +271,19 @@ namespace sistrip {
+        return true; // fast case: 2 bytes always fit an ADC (even if offset)
+      } else { // close to end
+        const uint16_t plusOneBitOffset = currentLocalBitOffset_+bitOffsetIncrement_;
+        const uint16_t plusOneWordOffset = currentWordOffset_ + plusOneBitOffset/8;
```
This magic number "8" appears rather often in this class.
It could be more conveniently defined somewhere, perhaps as a ```sizeof()``` a given data word

pieterdavid replied:
In this context it is the number of bits in a byte of raw data (`uint8_t`), so `sizeof` will not work (it gives the size of a type in bytes). I could define a constant, e.g. `constexpr uint16_t BITS_PER_BYTE = 8;`, but it would be only for readability.

perrotta replied:
That would help, indeed

pieterdavid committed:
sistrip::FEDBSChannelUnpacker: use BITS_PER_BYTE constant instead of 8

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-23417/5151

cmsbuild commented:
Pull request #23417 was updated. @perrotta, @civanch, @mdhildreth, @cmsbuild, @franzoni, @slava77, @fabiocos, @davidlange6 can you please check and sign again.

perrotta commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 81ef51b66f52a5f26cd8f74a462929350b8e51f9
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23417/28630/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/86a9c4c3025c89150d0623faf6034a298900750d
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23417/28630/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23417/28630/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23417/28630/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 12 differences found in the comparisons
 - DQMHistoTests: Total files compared: 31
 - DQMHistoTests: Total histograms compared: 2901984
 - DQMHistoTests: Total failures: 3
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2901791
 - DQMHistoTests: Total skipped: 190
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 30 files compared)
 - Checked 128 log files, 14 edm output root files, 31 DQM output files

fabiocos commented:
is there any news about this PR? I understand it was desired in pre6 and for possible tests during the TS1. Are there pending issues with it?

pieterdavid commented:
As far as I know there are no outstanding issues (at least I did not find any in the comparisons - but I may have missed something, I could not find all 12 differences in the logs).
We were asked to give an update on it in the reco meeting this afternoon, so we should know more soon. (I am waiting to submit the backport PR until this one is approved, but if it makes the release planning easier I can submit it at any time)

perrotta commented:
@pieterdavid : the new log message that appears in the 2011 workflows (and only in them) with this PR is the following:
```
%MSG-w SiStripRawToDigi:  SiStripRawToDigiModule:siStripDigis  11-Jun-2018 21:12:33 CEST Run: 165121 Event: 23609118
NULL pointer to FEDRawData for FED: id 87
Note: further warnings of this type will be suppressed (this can be changed by enabling debugging printout)
```
It is not obvious to me why it did not show up without this PR: could you please quickly investigate?

perrotta commented:
Answering myself: because it was previously protected by
```
if ( edm::isDebugEnabled() ) {
```
now removed

perrotta commented:
Then the only (minor) question I have is: does that NULL pointer to FED 87 in 2011 workflows represent an issue? Please notice that the 2011 HI workflow do not issue that warning, only the pp ones (MinBias and Cosmics). Maybe this is another instantiation of the bug in the packer that you were mentioning in https://github.com/cms-sw/cmssw/pull/23417#issuecomment-394783157

This is of course unrelated to your pull request, but it could be the right time to check it, and provide a fix if needed.

pieterdavid commented:
Thanks @perrotta , that message (and the corresponding summary of unpacker warnings at the end of the job) is indeed present in the three jobs with differences. I will investigate

perrotta commented:
+1
- SiStrip unpacker adapted for non-standard ZS modes as described: user tests show that the unpacked data reproduce the input ones
- No effect is visible in standard workflows, were those non-standard ZS modes are not at work
- Warning messages have also been reorganized and improved: this generate an extra warning (once per job) in the 2011 pp workflows. It has to be investigated, but outside the scope of this PR

civanch commented:
+1

pieterdavid commented:
Output of the packer-unpacker closure test (run with [this script](https://github.com/pieterdavid/SiStripLocalRecoDebugging/blob/master/testAllModes.zsh)), as requested during Friday's RECO meeting:
```
---> Testing VR packing on 5 events
Packet code :  cms.string('VIRGIN_RAW')
      5 Found 14710 dets with identical raw digis and 0 with differences
Packet code :  cms.string('VIRGIN_RAW8_BOTBOT')
      5 Found 14710 dets with identical raw digis and 0 with differences
Packet code :  cms.string('VIRGIN_RAW8_TOPBOT')
      5 Found 14710 dets with identical raw digis and 0 with differences
Packet code :  cms.string('VIRGIN_RAW10')
      5 Found 14710 dets with identical raw digis and 0 with differences
---> Testing ZS(lite) packing on 5 events
Readout mode:  cms.string('Zero suppressed')
Packet code :  cms.string('ZERO_SUPPRESSED')
Found 14690 dets with identical digis and 0 with differences
Found 14392 dets with identical digis and 0 with differences
Found 14687 dets with identical digis and 0 with differences
Found 14603 dets with identical digis and 0 with differences
Found 14247 dets with identical digis and 0 with differences
Readout mode:  cms.string('Zero suppressed')
Packet code :  cms.string('ZERO_SUPPRESSED8_BOTBOT')
Found 14690 dets with identical digis and 0 with differences
Found 14392 dets with identical digis and 0 with differences
Found 14687 dets with identical digis and 0 with differences
Found 14603 dets with identical digis and 0 with differences
Found 14247 dets with identical digis and 0 with differences
Readout mode:  cms.string('Zero suppressed')
Packet code :  cms.string('ZERO_SUPPRESSED8_TOPBOT')
Found 14690 dets with identical digis and 0 with differences
Found 14392 dets with identical digis and 0 with differences
Found 14687 dets with identical digis and 0 with differences
Found 14603 dets with identical digis and 0 with differences
Found 14247 dets with identical digis and 0 with differences
Readout mode:  cms.string('Zero suppressed')
Packet code :  cms.string('ZERO_SUPPRESSED10')
Found 14690 dets with identical digis and 0 with differences
Found 14392 dets with identical digis and 0 with differences
Found 14687 dets with identical digis and 0 with differences
Found 14603 dets with identical digis and 0 with differences
Found 14247 dets with identical digis and 0 with differences
Readout mode:  cms.string('Zero suppressed lite8')
Packet code :  cms.string('')
Found 14690 dets with identical digis and 0 with differences
Found 14392 dets with identical digis and 0 with differences
Found 14687 dets with identical digis and 0 with differences
Found 14603 dets with identical digis and 0 with differences
Found 14247 dets with identical digis and 0 with differences
Readout mode:  cms.string('Zero suppressed lite8 BotBot')
Packet code :  cms.string('')
Found 14690 dets with identical digis and 0 with differences
Found 14392 dets with identical digis and 0 with differences
Found 14687 dets with identical digis and 0 with differences
Found 14603 dets with identical digis and 0 with differences
Found 14247 dets with identical digis and 0 with differences
Readout mode:  cms.string('Zero suppressed lite8 TopBot')
Packet code :  cms.string('')
Found 14690 dets with identical digis and 0 with differences
Found 14392 dets with identical digis and 0 with differences
Found 14687 dets with identical digis and 0 with differences
Found 14603 dets with identical digis and 0 with differences
Found 14247 dets with identical digis and 0 with differences
Readout mode:  cms.string('Zero suppressed lite10')
Packet code :  cms.string('')
Found 14690 dets with identical digis and 0 with differences
Found 14392 dets with identical digis and 0 with differences
Found 14687 dets with identical digis and 0 with differences
Found 14603 dets with identical digis and 0 with differences
Found 14247 dets with identical digis and 0 with differences
```

pieterdavid commented:
On the "FED data missing" warnings (likely due to disabled FEDs): for the 1000.0 and 1001.0 workflows (run 165121) they are all about FED 87. For the 4.22 workflow (2011 cosmics, run 160960) the warning is there for three FEDs, including 87 (I tried to rerun with debugging on to get the other two IDs, but the das query doesn't return any results, and if I take the file list from the comparisons the cern xrootd server says that the file doesn't exist - indeed I do not see it on eos).

erikbutz commented:
The other two missing FED IDs should be 103 and 388 which are two FEDs that have been excluded for a long time already, FED 87 was a transitory problem that so happened to be present in the run that is used. 
In post-LS1 runs I would expect to get messages for FEDs missing for FEDs 59, 103 and 388 unless runs used there again happen to have a transitory problem.

mmusich commented:
Hello, just to confirm that also on the offline side, 
FED 87 (see it in the TrackerMap below: TIB L2, lower right corner)

![tmap_fed87](https://user-images.githubusercontent.com/5082376/41533188-39b15ffe-72fa-11e8-959b-2f1f39ef4dcb.png)

is seen in the list of components masked in the SiStrip cabling + excluded FEDs (from `RunInfo` O2O ) for run 165121: 

![mergedbadcomponentstkmap_run_165121](https://user-images.githubusercontent.com/5082376/41533333-b211bf66-72fa-11e8-8fec-c18cd57262de.png)

mmusich commented:
@pieterdavid 
> For the 4.22 workflow (2011 cosmics, run 160960) the warning is there for three FEDs, including 87 (I tried to rerun with debugging on to get the other two IDs, but the das query doesn't return any results, and if I take the file list from the comparisons the cern xrootd server says that the file doesn't exist - indeed I do not see it on eos).

for testing those workflows you can use the `--ibeos` option, that makes use of the global xrootd redirector when input files are not found at `T2_CH_CERN` :

- `runTheMatrix.py -l 4.22 --ibeos`

see discussion at https://github.com/cms-sw/cmssw/issues/22278

perrotta commented:
Thank you all for checking
Maybe the masked components could be taken into account and not trigger the warning message, then...

pieterdavid commented:
Thanks @mmusich , with `--ibeos` it worked. The other FEDs (besides 87) for 4.22 (run 160960) are 135 and 137. Since these are different than those @erikbutz mentioned, maybe they also had transient problems? The unpacker gets the list of FEDs to unpack from the cabling map (`SiStripFedCabling`).

mmusich commented:
Hi @pieterdavid , thanks for further the checks.
Both FED 135 and 137 are located in TID- Disk2:

![tmap_fed135_137](https://user-images.githubusercontent.com/5082376/41538620-091f821e-730c-11e8-981e-cbb4c725ac87.png)

and in my recollection they are not permanently out of the cabling. Indeed looks like a transient problem (also in this case bothe FEDs are seen in the offline masking for the `SiStripQuality` for run 160960):

![mergedbadcomponentstkmap_run_160960](https://user-images.githubusercontent.com/5082376/41538830-73cafa08-730c-11e8-96be-51680bb465aa.png)

fabiocos commented:
@pieterdavid @perrotta from the thread above I understand that the new messages appearing for 2011 data are related to a particular data period and understood, and should not represent a general problem of the PR. In any case, I consider that the legacy release to study Run1 remains the 53X.
If you confirm my understanding, I am ready to integrate this PR.

fabiocos commented:
+operations

the updates to the StandardSequences appear coherent with the rest of the PR

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar, @fabiocos (and backports should be raised in the release meeting by the corresponding L2)

perrotta commented:
Yes @fabiocos , I confirm that the messages are only exposed by this PR (as new warning messages are issued which did not exist before unless in some debug mode).
Therefore, whatever is the origin of those messages (and yes, according to the messages above that origin seems to be understood), it does not depend nor can be originated by this PR

fabiocos commented:
+1
